src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -71,6 +71,51 @@
  * &lt;module name="RightCurly"/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ *public class Test {
+ *
+ *  public void test() {
+ *
+ *    if (foo) {
+ *      bar();
+ *    }           // violation, right curly must be in the same line as the 'else' keyword
+ *    else {
+ *      bar();
+ *    }
+ *
+ *    if (foo) {
+ *      bar();
+ *    } else {     // OK
+ *      bar();
+ *    }
+ *
+ *    if (foo) { bar(); } int i = 0;
+ *                      // violation, statement is not allowed on same line after curly right brace
```

Comment:
violation comment should be on the same line where it is reported, desctiption comment can be on other line.
``` 
*    if (foo) { bar(); } int i = 0;  // violation
*                  // ^^^ statement is not allowed on same line after curly right brace
```

Reply:
done